src/protocol/IERC20Bridge.sol

```diff
@@ -0,0 +1,109 @@
+    /// @param id The initialization id
+    /// @param initialization The token initialization data
+    event TokenInitialized(bytes32 indexed id, TokenInitialization initialization);
+
+    /// @dev Emitted when a token initialization is proven on the destination chain.
+    /// @param id The initialization id
+    /// @param initialization The token initialization data
+    /// @param deployedToken The address of the deployed bridged token
+    event TokenInitializationProven(
+        bytes32 indexed id, TokenInitialization initialization, address indexed deployedToken
+    );
+
+    /// @dev Emitted when a deposit is made.
+    /// @param id The deposit id
+    /// @param deposit The ERC20 deposit
+    event DepositMade(bytes32 indexed id, ERC20Deposit deposit);
+
+    /// @dev Emitted when a deposit is claimed.
+    /// @param id The deposit id
+    /// @param deposit The claimed ERC20 deposit
+    event DepositClaimed(bytes32 indexed id, ERC20Deposit deposit);
+
+    /// @dev Failed to claim the deposit.
+    error FailedClaim();
+
+    /// @dev A deposit was already claimed.
+    error AlreadyClaimed();
+
+    /// @dev Token initialization has already been proven.
+    error InitializationAlreadyProven();
+
+    /// @dev Whether the deposit identified by `id` has been claimed or cancelled.
+    /// @param id The deposit id
+    function processed(bytes32 id) external view returns (bool);
+
+    /// @dev Whether a token initialization has been proven (on destination chain).
+    /// @param id The initialization id
+    function isInitializationProven(bytes32 id) external view returns (bool);
+
+    /// @dev Get the deployed token address for an original token (on destination chain).
+    /// @param originalToken The original token address
+    function getDeployedToken(address originalToken) external view returns (address);
+
+    /// @dev Token initialization identifier.
+    /// @param tokenInit The token initialization struct
+    function getInitializationId(TokenInitialization memory tokenInit) external pure returns (bytes32 id);
+
+    /// @dev ERC20 Deposit identifier.
+    /// @param erc20Deposit The ERC20 deposit struct
+    function getDepositId(ERC20Deposit memory erc20Deposit) external pure returns (bytes32 id);
+
+    /// @dev Initializes a token for bridging by reading its metadata and sending a signal.
+    /// @param token The ERC20 token address to initialize
+    function initializeToken(address token) external returns (bytes32 id);
+
+    /// @dev Proves a token initialization from the source chain and deploys the bridged token.
+    /// @param tokenInit The token initialization data
+    /// @param height The height of the checkpoint on the source chain
+    /// @param proof Encoded proof of the initialization signal
+    function deployCounterpartToken(TokenInitialization memory tokenInit, uint256 height, bytes memory proof)
+        external
+        returns (address deployedToken);
```

Comment:
I think we should update all functions/structs/events to consistently describe these operations as (1) logging the token description and (2) deploying the counterpart token. For instance:

- I would change `TokenInitialization` and `TokenInitialized` to `TokenDescription` and `TokenDescriptionRecorded` or similar
- I don't think we need an `isInitializationProven` function if it's just checking that a particular `id` is `processed`. Anyone who knows what that ID is will know that it applies to the token description
- `TokenInitializationProven` should be `CounterpartTokenDeployed`  (or `BridgedTokenDeployed`) since that is the actual action taken by the function. Similarly, the `InitializationAlreadyProven` error should be `CounterpartTokenAlreadyDeployed`

Reply:
https://github.com/OpenZeppelin/minimal-rollup/pull/146/commits/6361902e5545fb5d6064c37621bb64586c240fe2